src/classes/BDI_DataImportBatch_SEL.cls

```diff
@@ -51,8 +52,46 @@ public with sharing class BDI_DataImportBatch_SEL {
                     Donation_Matching_Behavior__c,
                     Donation_Matching_Implementing_Class__c,
                     Donation_Matching_Rule__c,
-                    Donation_Date_Range__c
+                    Donation_Date_Range__c,
+                    Process_with_Scheduled_Job__c
             FROM DataImportBatch__c
             WHERE Id in :setBatchId];
     }
+
+    /*******************************************************************************************************
+    * @description queries the database for the Data Import Batch record with the specified Id
+    * @param batchId The batch to query for.
+    * @return DataImportBatch__c
+    */
+    public static DataImportBatch__c selectById(Id batchId) {
+        list<DataImportBatch__c> listBatches = selectByIds(new set<Id>{batchId});
+        return listBatches[0];
+    }
+
+    /*******************************************************************************************************
+    * @description queries the database for all Data Import Batch records that are marked with Scheduled
+    * processing, and have DI records to import.
+    * @return list<Id>
+    */
+    public static list<Id> selectWithScheduledProcessing() {
+        List<Id> listIdBatches = new List<Id>();
+
+        for (DataImportBatch__c diBatch : [SELECT
+            Id,
```

Comment:
Two problems with this method.

1. in its current form, this SOQL is non-selective.  QueryPlan says it's a tablescan with cost of 1.96.  There is no index on the Process checkbox. 

2. if we decide to support a DataImportBatchName text field on the import object, then we can't use the subquery.  We'd need to find all of the batches who have unimported records (by either the batch lookup or batch name), and then query against batches that have Process checked and are in this set of batches with records to be imported.  Unfortunately, getting all the unique batch names or batch Id's requires using GROUP BY queries, and those are not going to be selective.  Just getting the unique batch Id's with:

Select NPSP_Data_Import_Batch__c from DataImport__c where NPSP_Data_Import_Batch__c <> null AND Status__c <> 'Imported' GROUP BY NPSP_Data_Import_Batch__c

is a Tablescan with cost 0.883

if we try to get both fields in a single GROUP BY query with something like:

SELECT NPSP_Data_Import_Batch__c, NPSP_Data_Import_Batch_Name__c FROM DataImport__c WHERE Status__c != 'Imported' GROUP BY NPSP_Data_Import_Batch__c, NPSP_Data_Import_Batch_Name__c 

we get a tablescan at cost 2.68

So I'm not sure how to tackle this to handle LDV orgs where query selectiveness is key.  Thoughts anyone? @njjc @joy-sfdo @force2b @RadSFDeveloper @jjbennett @cdcarter @judisohn @szeitz